### PR TITLE
Add resource/min_gc_num_siblings_deleted blockchain param

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Add resource/min_gc_num_siblings_deleted blockchain param
- Merge removeOldReceipts() back to recordReceipt()
- Fix tests

Related issues:
- https://github.com/ainblockchain/ain-blockchain/issues/817
- https://github.com/ainblockchain/ain-blockchain/issues/478

### CODE REVIEW

File: db/state-util.js [hunk 1]
```diff
@@ -536,7 +537,8 @@ function isValidConfigTreeRecursive(
     subtreePath.push(label);
     const subtree = stateTreeObj[label];
     if (label === configLabel) {
-      const isValidConfig = stateConfigValidator([...treePath, ...subtreePath], subtree);
+      const newParams = Object.assign({}, params, { configPath: [...treePath, ...subtreePath]});
```
Comment (liayoo) [inline db/state-util.js:540]: minor) a space after the array?

File: db/index.js [hunk 2]
```diff
@@ -1471,17 +1475,13 @@ class DB {
     //               and collectFee().
     const parsedPath = CommonUtil.parsePath(receiptPath);
     this.writeDatabase([PredefinedDbPaths.VALUES_ROOT, ...parsedPath], receipt);
-  }
 
-  removeOldReceipts() {
```
Comment (liayoo) [inline db/index.js:1476]: this is nice!! thank you~

**Review by liayoo — APPROVED** [review]
LGTM with a minor comment. Thanks!

File: db/state-util.js (hunk 1 above)
Comment (platfowner) [reply db/state-util.js]: done, thanks!

**Pull request merged**